e2eAIOK/common/trainer/model_builder.py

```diff
@@ -15,38 +15,40 @@ def __init__(self, cfg):
         super().__init__()
         self.cfg = cfg
     
-    '''
-    ceate the model for training or evluation
-    '''
+    def _pre_process(self):
+        self.logger = logging.getLogger('Trainer')
+        self.logger.info("building model")
+
+    def _init_model(self):
+        """
+            create model
+        """
+        raise NotImplementedError("_init_model is abstract.")
+    
+    def _post_process(self):
+        """
+            post work after create model
+        """
+        print(f"model created: {self.model}")
+
     def create_model(self):
```

Comment:
1. Can we merge create_model() and __init__() together, in that way we can create model when initialize the Builder.
2. We can add a parameter "model" to __init__() to load extra models.

Reply:
For the 2nd suggestion, maybe we can do it like "__init__(self, *kargs)" for extra arguments. Thx.

Reply:
But we need to use it in the next step, such as:
> if model is not None:
>       self.model = model
> else:
>      self.model = _init_model()

I this case, I suggest model can work as a normal parameter.

Reply:
add "model" parameter in init() to load pre-trained model